pxr/imaging/hdSt/materialXFilter.h

```diff
@@ -35,18 +44,68 @@ struct HdSt_MxShaderGenInfo {
     bool bindlessTexturesEnabled;
 };
 
+/// Encapsulates the input data for MaterialX codegen for an individual shader.
+///
+class HdSt_MaterialXGeneratorTask final
+{
+public:
+    /// Modifies the input filter task - stores mappings between original and
+    /// anonymized Sdf paths.
+    ///
+    HdSt_MaterialXGeneratorTask(
+        HdSt_MaterialFilterTaskSharedPtr filterTask,
+        SdfPath const& materialPath,
+        Hgi const& hgi);
```

Comment:
I know you're improving on the status quo here (which is to pass in resource registry and pull hgi out of that), but it would be cleaner to just pass in the relevant info, namely bindless support and shading language.